prusti-tests/tests/verify/pass/equality/simple-enum.rs

```diff
@@ -0,0 +1,16 @@
+use prusti_contracts::*;
+
+#[derive(Clone,PartialEq,Eq)]
+enum A {
+    ANone(u32),
+    ASome(i32),
+}
+
+#[requires(_x == _y)]
+#[ensures(result == 17)]
+fn test(_x: &A, _y: &A) -> i32 {
+    17
```

Comment:
I assume it works, but could we have tests (both passing and failing) that also call these `test` methods with some instances of the enums?

Reply:
I see, this one will not work because we can't encode reference copy types.

Reply:
> we can't encode reference copy types

Why? Shared references should not be a problem. Why3 does that.

Reply:
Maybe it's just in specific circumstances. The error when I tried this is the following:

https://github.com/viperproject/prusti-dev/blob/95409028b9acbd77fa5ca516ad0d2e4f1d21b298/prusti-viper/src/encoder/procedure_encoder.rs#L4960-L4969

Reply:
Something is strange here. If I recall correctly, this error is only for cases when reference-typed fields are used, which should not be the case here.